projects/step-frontend/src/lib/modules/_common/components/execution-link/execution-link.component.ts

```diff
@@ -0,0 +1,22 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { Execution, ExecutionsService } from '@exense/step-core';
+
+@Component({
+  selector: 'step-execution-link',
+  templateUrl: './execution-link.component.html',
+  styleUrls: ['./execution-link.component.scss'],
+})
+export class ExecutionLinkComponent implements OnChanges {
+  @Input() executionId?: string;
+  @Input() executionDescription?: string;
+
+  constructor(private _executionService: ExecutionsService) {}
+
+  ngOnChanges(): void {
+    if (this.executionId && !this.executionDescription) {
+      this._executionService.get3(this.executionId!).subscribe((ex: Execution) => {
+        this.executionDescription = ex?.description;
+      });
```

Comment:
This might be executed unexpectedly and request resources from the BE. 

You should prbably check if changes.executionId.currentValue !== changes.executionId.previousValue or changes.executionId.executionDescription!== changes.executionId.executionDescription before doing the request.

see:
[ngOnChanges(changes: SimpleChanges): void](https://angular.io/api/core/OnChanges#ngOnChanges)

Reply:
see https://github.com/exense/step-frontend/pull/59#discussion_r924244647